app/(playground)/p/[agentId]/beta-proto/artifact/server-actions.ts

```diff
@@ -71,38 +70,28 @@ ${sourcesToText(sources)}
 				prompt: params.userPrompt,
 				schema: artifactSchema,
 				onFinish: async (result) => {
-					const duration = performance.now() - startTime;
-					const measurementScope = await getCurrentMeasurementScope();
-					const isR06User = await isRoute06User();
-					generation.end({
-						output: result,
-					});
-
-					logger.info(
-						{
-							externalServiceName: "openai",
-							tokenConsumed: {
-								input: result.usage.promptTokens,
-								output: result.usage.completionTokens,
-							},
-							duration,
-							measurementScope,
-							isR06User,
+					await withTokenMeasurement(
+						logger,
+						async () => {
+							generation.end({ output: result });
+							await lf.shutdownAsync();
+							waitUntil(
+								new Promise((resolve) =>
+									setTimeout(
+										resolve,
+										Number.parseInt(
+											process.env.OTEL_EXPORT_INTERVAL_MILLIS ?? "1000",
+										) +
+											Number.parseInt(
+												process.env.WAITUNTIL_OFFSET_MILLIS ?? "0",
```

Comment:
added offset to make backend Lambda function to wait telemetry data to be exported